pad_server/src/gpio_actuator.c

```diff
@@ -0,0 +1,56 @@
+// BEWARE, this file is expluded from Linux compilation
+// because the of the missing nuttx header file
+
+#include "gpio_actuator.h"
+#include "stdio.h"
+#include <errno.h>
+#include <fcntl.h>
+#include <nuttx/ioexpander/gpio.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+int gpio_actuator_on(actuator_t *act) {
+    // TODO: change the device to be actuator specific
+    int fd = open("/dev/gpio12", O_RDWR);
+    if (fd < 0) {
+        fprintf(stderr, "Failed to open gpio with err %d\n", errno);
+        return -1;
+    }
+
+    int err = ioctl(fd, GPIOC_WRITE, true);
+    if (err < 0) {
+        fprintf(stderr, "Failed to communicate via ioctl with err %d\n", errno);
+        return -1;
```

Comment:
These functions should return an `errno` variable. So if `close` fails, instead of returning `-1`, return a descriptive error code from `errno.h`.